### PR TITLE
fix: Drop down multiselect usage fixes for query forms

### LINKED ISSUE #11464: [Bug]: Multiselect issue in DropDownControl for query editor forms
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When we pass the prop of `isMultiSelect` to a drop-down in editor JSON, the UI fails as the `DropDownControl` file is not designed to handle multiple options.

The expected behaviour is that the file should be able to seamlessly handle both multi-select and single select files.

### Steps To Reproduce

1. Edit any editor.json of any plugin and add `isMultiSelect='true'` to any drop-down component.
2. Load the form in the app
3. The app will give an error state.

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description

The drop-down control for query forms had issues for multi-select options. The PR adds the following changes
1. The selected value is now accepting strings or an array of strings.
2. The `onSelect` and `removeSelect` functions added.
3. Special handling for when multi-select is enabled.
4. Added search option to the dropdown.
5. Updated usage of the Projection component to point to the dropdown file.

Fixes #11464 

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: fix/11464-drop-down-multiselect-usage-update 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 55.7 **(-0.02)** | 37.03 **(0.02)** | 35.02 **(0.01)** | 56.04 **(-0.02)**
 :red_circle: | app/client/src/components/formControls/DropDownControl.tsx | 23.88 **(-16.12)** | 2.11 **(-1.97)** | 12.5 **(-4.17)** | 22.73 **(-15.73)**
 :red_circle: | app/client/src/utils/FormControlRegistry.tsx | 38.18 **(-1.11)** | 100 **(0)** | 5.26 **(0)** | 38.18 **(-1.11)**
 :green_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.94 **(0.23)** | 41.67 **(0.84)** | 36.21 **(0)** | 56.99 **(0.25)**
 :x: | ~~app/client/src/components/formControls/ProjectionSelectorControl.tsx~~ | ~~36.36~~ | ~~2.86~~ | ~~11.11~~ | ~~35.71~~</details>

### CODE REVIEW

File: app/client/src/components/formControls/DropDownControl.tsx [hunk 1]
```diff
@@ -53,32 +53,81 @@ function renderDropdown(
...
+      }) || [];
   }
+  // Function to handle selction of options
+  const onSelectOptions = (value: string | undefined) => {
+    if (value) {
+      if (props.isMultiSelect) {
+        if (Array.isArray(selectedValue) && !selectedValue.includes(value))
+          (selectedValue as string[]).push(value);
+        else {
+          selectedValue = [selectedValue as string, value];
```
Comment (AmanAgarwal041) [inline app/client/src/components/formControls/DropDownControl.tsx:92]: If selectedValue is an array but it does not have the clicked values inside it, then will the else condition not create a 2d array ?

File: app/client/src/components/formControls/DropDownControl.tsx [hunk 2]
```diff
@@ -53,32 +53,81 @@ function renderDropdown(
...
+  const onRemoveOptions = (value: string | undefined) => {
+    if (value) {
+      if (props.isMultiSelect) {
+        if (Array.isArray(selectedValue) && selectedValue.includes(value))
+          (selectedValue as string[]).splice(
+            (selectedValue as string[]).indexOf(value),
+            1,
+          );
+        else {
+          selectedValue = "";
```
Comment (AmanAgarwal041) [inline app/client/src/components/formControls/DropDownControl.tsx:109]: If the selectedValue is an array and the value is not present inside it, the else condition will change the array to empty string.

**Review by Irongade — APPROVED** [review]

**Pull request merged**